### PR TITLE
Adding unit tests to methods of pod's format

### PR DESCRIPTION
**Pull request by chentao1596**
What this PR does / why we need it:

Add unit test cases, thank you!

### CODE REVIEW

**Comment by xiangpengzhao** [comment]
/ok-to-test

It's strange it wasn't triggered.

**Comment by zhangxiaoyu-zidif** [comment]
@xiangpengzhao this pr may forget to modify BUILD file. Alright?

**Comment by xiangpengzhao** [comment]
I think so.
@chentao1596 please run `hack/update-bazel.sh` to see if there are BUILD files updated. If so, please commit them.

**Comment by chentao1596** [comment]
@zhangxiaoyu-zidif 
@xiangpengzhao 
Yes, I have updated it, thank you!

**Comment by chentao1596** [comment]
@feiskyer @zhangxiaoyu-zidif @xiangpengzhao 
PTAL, thank you!

File: pkg/kubelet/util/format/pod_test.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+			// Create test Ojbect
+			fakePod := fakeCreatePod(testCase.podName, testCase.podNamesapce, testCase.podUID)
+
+			realPod := Pod(fakePod)
+			if realPod != testCase.expectedValue {
+				t.Fatalf("Failed to format pod: Get %s, but want %s", realPod, testCase.expectedValue)
+			}
+
+			realPodDesc := PodDesc(fakePod.Name, fakePod.Namespace, fakePod.UID)
+			if realPodDesc != testCase.expectedValue {
```
Comment (feiskyer) [inline pkg/kubelet/util/format/pod_test.go]: Please use `assert.Equal` for such comparison (import `github.com/stretchr/testify/assert`)

**Review by feiskyer — CHANGES_REQUESTED** [review]

File: pkg/kubelet/util/format/pod_test.go (hunk 1 above)
Comment (chentao1596) [reply pkg/kubelet/util/format/pod_test.go]: I have tried to change it with 'assert', it look like nice. thank you!

File: pkg/kubelet/util/format/pod_test.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+		expectedValue string
+	}{
+		{"input_nil_case", nil, ""},
+		{"input_empty_case", []*v1.Pod{}, ""},
+		{"input_length_one_case", []*v1.Pod{pod1}, "pod1_default(551f5a43-9f2f-11e7-a589-fa163e148d75):DeletionTimestamp=2017-09-26T14:37:50Z"},
+		{"input_length_more_than_one_case", []*v1.Pod{pod1, pod2}, "pod1_default(551f5a43-9f2f-11e7-a589-fa163e148d75):DeletionTimestamp=2017-09-26T14:37:50Z, pod2_default(e84a99bf-d1f9-43c2-9fa5-044ac85f794b):DeletionTimestamp=2017-09-26T14:37:50Z"},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.caseName, func(t *testing.T) {
```
Comment (feiskyer) [inline pkg/kubelet/util/format/pod_test.go]: `t.Run` is not required, please remove it

Comment (chentao1596) [reply pkg/kubelet/util/format/pod_test.go]: It's done, PTLA, thank you, ^_^!

File: pkg/kubelet/util/format/pod_test.go [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+limitations under the License.
+*/
+
+package format
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
```
Comment (xiangpengzhao) [inline pkg/kubelet/util/format/pod_test.go]: please add a line between k8s import and other url/pkg import.

File: pkg/kubelet/util/format/pod_test.go [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+		{"input_length_more_than_one_case", []*v1.Pod{pod1, pod2}, "pod1_default(551f5a43-9f2f-11e7-a589-fa163e148d75), pod2_default(e84a99bf-d1f9-43c2-9fa5-044ac85f794b)"},
+	}
+
+	for _, testCase := range testCases {
+		realPods := Pods(testCase.pods)
+		assert.Equalf(t, testCase.expectedValue, realPods, "Failed to test: %s", testCase.caseName)
+	}
+}
+
+func TestPodsWithDeletiontimestamps(t *testing.T) {
```
Comment (xiangpengzhao) [inline pkg/kubelet/util/format/pod_test.go]: I prefer `TestPodsWithDeletionTimestamps`. (Also rename the original func to `PodsWithDeletionTimestamps`)

**Review by xiangpengzhao — CHANGES_REQUESTED** [review]
two nits.

File: pkg/kubelet/util/format/pod_test.go (hunk 4 above)
Comment (chentao1596) [reply pkg/kubelet/util/format/pod_test.go]: It should be more standardized.
But, they are not the same contents, so i split them into different commits, thank you!

**Comment by chentao1596** [comment]
@feiskyer @xiangpengzhao hi, PTAL, thank you!

**Review by feiskyer — APPROVED** [review]
/lgtm thanks

**Comment by chentao1596** [comment]
@feiskyer @xiangpengzhao @vishh 
Hi, i update commits because of the ask of rebase, please help to see it again, thank you!

**Comment by xiangpengzhao** [comment]
/assign @vishh @yujuhong 
for an approval with `/approve no-issue`.

**Comment by chentao1596** [comment]
@feiskyer @xiangpengzhao hi, because of the conflict, i have rebase the branch, PTAL, thank you

**Pull request merged**